### ISSUE TITLE
logging-fstring-interpolation not raised on function parameter hinted as logging.Logger

### ISSUE THREAD

**Issue by ringohoffman** [pylint-dev/pylint#8781]
### Bug description

If I log to the root logger directly or if within a scope I call `logging.getLogger` and then log to that `Logger`, I will get the `logging-fstring-interpolation` warning, but if I have a function parameter that is hinted as a `logging.Logger`, I will not get the `logging-fstring-interpolation` warning on that logger:

```python
"""A demonstration of a missing logging-fstring-interpolation warning."""
import logging

MSG = "we used an f-string in a logging statement"
logging.error(f"Something bad happened: {MSG}")

file_logger = logging.getLogger(__name__)
file_logger.error(f"Something bad happened: {MSG}")


def my_function(logger: logging.Logger) -> None:
    """A function that accepts a logger."""
    logging.error(f"Something bad happened: {MSG}")

    function_logger = logging.getLogger(MyClass.__name__)
    function_logger.error(f"Something bad happened: {MSG}")

    logger.error(f"Something bad happened: {MSG}")  # no logging-fstring-interpolation warning


class MyClass:  # pylint: disable=too-few-public-methods
    """A class with a method that accepts a logger."""

    def __init__(self, logger: logging.Logger) -> None:
        logging.error(f"Something bad happened: {MSG}")

        class_logger = logging.getLogger(MyClass.__name__)
        class_logger.error(f"Something bad happened: {MSG}")

        logger.error(f"Something bad happened: {MSG}")  # no logging-fstring-interpolation warning

```

### Configuration

```ini
N/A
```

### Command used

```shell
`pylint a.py`
```

### Pylint output

```shell
$ pylint a.py
************* Module a
a.py:5:0: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:8:0: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:13:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:16:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:25:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:28:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
```

### Expected behavior

I expect to get a `logging-fstring-interpolation` warning on the `logger.error` statements too.

### Pylint version

```shell
astroid==2.15.4
pylint==2.17.4
```
```


### OS / Environment

Darwin

### Additional dependencies

_No response_

**Comment by Pierre-Sassoulas** [comment]
I think it depends on #4813 and is not going to be solved as long as it's not done, so I'm hesitating to close as duplicate even if it seem it's two very different issues.

**Comment by ringohoffman** [comment]
Yes, so it seems. Kind of a shocking to realize that `astroid` doesn't currently use type hints to infer the origin of an expression. It greatly diminishes the usefulness of this warning (and others as well I'm sure) in an object oriented setting.

**Comment by Pierre-Sassoulas** [comment]
astroid was created in 2003 and typing did not exists at the time :)

**Comment by fenchu** [comment]
Just drop it I disable W1203 everywhere:
![image](https://github.com/pylint-dev/pylint/assets/11349883/f9e6ed0b-3d94-465f-a683-890ecf5b83e3)
